cpp/src/arrow/python/common.h

```diff
@@ -64,12 +64,12 @@ inline Status CheckPyError(StatusCode code = StatusCode::UnknownError) {
 // This function can set a Python exception.  It assumes that T has a (cheap)
 // default constructor.
 template <class T>
-T GetResultValue(Result<T>& result) {
+T GetResultValue(Result<T> result) {
```

Comment:
Do you want a rvalue instead? If this is a C++/Python bridge limitation, please document via comment.

Reply:
You mean a `Result<T>&&`? Why not, but would it bring something (forbid passing a lvalue perhaps)?

Reply:
That would indeed forbid passing an lvalue.

Reply:
Ok. TBH, I'm not sure we care about that.

Reply:
I meant universal reference instead of rvalue reference. Right now it always copy. This might get tricky to implement due to the requirement of returning even in the case of a failure, @bkietz.

Reply:
It doesn't copy - just move - if the argument is a rvalue, right?

Reply:
The difference it brings is that it can support lvalue without expliciting the move, might not be worth the effort.

Reply:
@fsaintjacques That would be true if `Result` was a template parameter, but it's a concrete class here. `Result<T>&&` simply wouldn't match a lvalue, AFAIU.

See https://isocpp.org/blog/2012/11/universal-references-in-c11-scott-meyers and in particular the example:
```c++
template<typename T>
void f(std::vector<T>&& param);  // here, “&&” means rvalue reference
 
template<typename T>
void f(T&& param);               // here, “&&”does not mean rvalue reference
```